src/helper/securityQuestion.ts

```diff
@@ -0,0 +1,152 @@
+import { decrypt, encrypt, EncryptedMessage, getPubKeyECC, StringifiedType } from "@tkey-mpc/common-types";
+import { keccak256 } from "@toruslabs/torus.js";
+import BN from "bn.js";
+
+import { FactorKeyTypeShareDescription, TssFactorIndexType } from "../constants";
+import { Web3AuthMPCCoreKit } from "../mpcCoreKit";
+
+export class TssSecurityQuestionStore {
+  shareIndex: string;
+
+  associatedFactor: EncryptedMessage;
+
+  questions: string;
+
+  constructor(shareIndex: string, associatedFactor: EncryptedMessage, questions: string) {
+    this.shareIndex = shareIndex;
+    this.associatedFactor = associatedFactor;
+    this.questions = questions;
+  }
+
+  static fromJSON(json: StringifiedType) {
+    const { shareIndex, encrypted, questions } = json;
+    return new TssSecurityQuestionStore(shareIndex, encrypted, questions);
+  }
+
+  toJSON(): StringifiedType {
+    return {
+      shareIndex: this.shareIndex,
+      encrypted: this.associatedFactor,
+      questions: this.questions,
+    };
+  }
+}
+
+export class TssSecurityQuestion {
+  StoreDomainName = "tssSecurityQuestion";
+
+  async setSecurityQuestion(
+    mpcCoreKit: Web3AuthMPCCoreKit,
+    question: string,
+    answer: string,
+    description?: Record<string, string>,
+    factorKey?: string
+  ) {
+    if (!mpcCoreKit.tKey) {
+      throw new Error("Tkey not initialized, call init first.");
+    }
+    if (!question || !answer) {
+      throw new Error("question and password are required");
```

Comment:
the answer might not be a password, Question can be like: "What was your first pet name?", in that case we should not throw errors specific to password, i think we should take a extra optional param named: `isPassword` in `setSecurityQuestion`  and only do password validation if that is set to true.